src/Dhall/Import.hs

```diff
@@ -321,6 +320,38 @@ instance Show MissingEnvironmentVariable where
         <>  "\n"
         <>  "↳ " <> Text.unpack name
 
+-- | Wrapper type for all Exceptions related to Imports
+data MissingImport = forall e. Exception e => MissingImport e
+  deriving (Typeable)
+
+instance Exception MissingImport
+
+instance Show MissingImport where
+  show (MissingImport e) = show e
+
+-- | List of Exceptions we encounter while resolving Import Alternatives
+newtype MissingImports = MissingImports [MissingImport]
+  deriving (Typeable)
```

Comment:
You don't need to derive `Typeable` any longer.  I recently learned that it's automatically derived for all Haskell types without having to specify this

Reply:
Interesting. IIRC in our Haskell projects at work we have to derive it, but it probably depends on GHC version and/or extensions, I'll read up on this.